plutus-core/plutus-ir/test/lets/letInLet.golden

```diff
@@ -1,10 +1,7 @@
 (program
   1.0.0
   [
-    (lam unitval2_i0 (all a_i0 (type) (fun a_i1 a_i1)) unitval2_i1)
```

Comment:
Seems that let-floating has triggered the inliner.

Reply:
I thought it was because of the deadcode elimination.

Reply:
I *think* the floating out brings new opportunities for the deadcode eliminator that runs later to eliminate dead bindings.

https://github.com/input-output-hk/plutus/blob/53ab40d937a6d2c4b23c03fe4bbed463e8b3b86e/plutus-core/plutus-ir/src/PlutusIR/Compiler.hs#L200

Because the `PlutusIR.Analysis.Dependencies.depgraph` does not use free variables but some kind of scoping, then the scoping has changed after floating which results to new SCCs of depgraph being generated, which then reveals more dead bindings.

Reply:
Actually I take it back, it is indeed what @jakzale said that the inliner is triggered at a later step. `unitval2` is not dead.

https://github.com/input-output-hk/plutus/blob/53ab40d937a6d2c4b23c03fe4bbed463e8b3b86e/plutus-core/plutus-ir/src/PlutusIR/Compiler.hs#L202